src/main/golo/assertions.golo

```diff
@@ -0,0 +1,293 @@
+----
+Display results of assertions
+----
+function displayTestsReport = {
+  println("tests: passed:"+report: passed()+" failed:"+report: failed())
+}
+----
+get the results of assertions
+
+- *return* results of assertions (struct testsReport)
+----
+function getTestsReport = -> report
+
+# We need to know if we can use ansi codes
+let not_windows = likelySupported()
+
+----
+Helper to decompose a stacktrace line about error with a Golo script
+
+- *param* `path` the path to parse (String)
+----
+local function getDetailsOfGoloError = |stackTraceLine| {
+  let split = stackTraceLine: toString(): split("\\(")
+  let moduleNameAndMethod = split: get(0)
+  let fileName = split: get(1): split(":"): get(0)
+  let lineNumber = Integer.valueOf(split: get(1): split(":"): get(1): split("\\)"): get(0))
+
+  if(not_windows) { fg_blue() }
+
+  if(goloFiles: size(): equals(0)) {
+    filesDiscover(currentDir(), "golo", goloFiles, currentDir())
+  }
+
+  goloFiles: filter(|filePathName| {
+    return filePathName: endsWith(fileName)
+  }): each(|filePathName| {
+    let lines = java.nio.file.Files.readAllLines(
+      java.nio.file.Paths.get(currentDir()+filePathName),
+      java.nio.charset.Charset.forName("UTF-8")
+    )
+    let moduleNameInSourceCode = lines:get(0): split("module "): get(1): trim()
+    if(moduleNameAndMethod: startsWith(moduleNameInSourceCode+".")) {
+      println("  line number: " + lineNumber + ": " + lines: get(lineNumber - 1) + " ...")
+    }
+  })
+}
+
+----
+Helper to test predicate
+
+- *param* `predicate` this is a closure (FunctionReference)
+- *param* `allStackTrace` if true, the complete Java StackTrace is displayed, if false, only Golo stuff is displayed (Boolean)
+- *param* `getSuccessMessage` this is a closure that return a string (FunctionReference)
+- *param* `onSuccess` this is the callBack if success (FunctionReference)
+- *param* `getErrorMessage` this is a closure that return a string (FunctionReference)
+- *param* `exitWhenFailed` if true, then the program aborts at the first error (Boolean)
+- *param* `onError` this is the callBack if error (FunctionReference)
+----
+function assert = |predicate, allStackTrace, getSuccessMessage, onSuccess, getErrorMessage, exitWhenFailed, onError| {
+
+  if(not_windows) { reset() }
+  try {
+    # run the predicate
+    let result = predicate()
+    # if predicate is false then display error message
+    # and throw exception
+    if(result isnt true) {
+      if(not_windows) { fg_red() }
+      let errorMessage = getErrorMessage()
+      println(errorMessage)
+      report: failed(report: failed()+1)
+      throw Exception(errorMessage)
+    } else {
+      # predicate is verified, then display success message
+      # and execute success callback if exists
+      if(not_windows) { fg_green() }
+      println(getSuccessMessage())
+      if(not_windows) { reset() }
+      report: passed(report: passed()+1)
+      Option(onSuccess): either(|callBack| -> callBack(result), {})
```

Comment:
The main purpose of `Option` is to ease composition/chaining of *pure* functions that can fail. Here you use it as a control structure to apply side effect conditionally.